chef.py

```diff
@@ -0,0 +1,231 @@
+    doc = get_parsed_html_from_url(book_url)
+    destination = tempfile.mkdtemp()
+
+    def download_assets(selector, attr, middleware=None):
+        nodes = doc.select(selector)
+        for i, node in enumerate(nodes):
+            url = make_fully_qualified_url(node[attr])
+            filename = "%s_%s" % (i, os.path.basename(url))
+            node[attr] = filename
+            download_file(url, destination, request_fn=make_request, filename=filename, middleware_callbacks=middleware)
+
+    def js_middleware(content, url, **kwargs):
+        # Polyfill window.localStorage as iframes can't access localStorage.
+        return content.replace("window.localStorage",
+                "({setItem: function(){}, removeItem: function(){}})")
+
+    # Download all static assets.
+    # TODO(davidhu): Also download fonts referenced in http://www.africanstorybook.org/css/app.css
+    download_assets("img[src]", "src")  # Images
+    download_assets("link[href]", "href")  # CSS
+    download_assets("script[src]", "src", middleware=js_middleware) # JS
+
+    # Download all background images, e.g. <div style="background-image:url()">
+    # (africanstorybook.org uses these for the main picture found on each page
+    # of the storybook.)
+    bg_img_nodes = doc.select("div[style*=\"background-image:url(\"]")
+    for i, node in enumerate(bg_img_nodes):
+        style = node["style"]
+        match = BG_IMG_RE.search(style)
+        if not match:
+            continue
+
+        url = make_fully_qualified_url(match.group(1))
+        filename = "%s_%s" % (i, os.path.basename(url))
+        node["style"] = BG_IMG_RE.sub("background-image:url(%s)" % filename, style)
+        download_file(url, destination, request_fn=make_request, filename=filename)
+
+    # Hide the African Storybook header nav bar.
+    header = doc.select_one("#headerBar")
+    if header:
+        header["style"] = "display: none;"
+
+    with open(os.path.join(destination, "index.html"), "w") as f:
+        f.write(str(doc))
+
+    zip_path = create_predictable_zip(destination)
+
+    source_id = parse_qs(urlparse(book_url).query)['id'][0]
+    raw_title = doc.select_one("head title").text
+    title = raw_title.replace('African Storybook -', '').strip()
+
+    # Extract the description from the "Share to Facebook" text.
+    # TODO(davidhu): Find a more robust way to get the book description -- the
+    # description doesn't seem to exist in another way in the page source.
+    send_facebook_node = doc.select_one("a[onclick*=\"sendFacebook\"]")
+    send_facebook_text = send_facebook_node["onclick"]
+    match = SEND_FACEBOOK_RE.search(send_facebook_text)
+    if match:
+        description = match.group(1).strip("'\" ")
+    else:
+        raise Exception("Could not extract book description from Share to Facebook text: %s" % send_facebook_text)
+
+    return nodes.HTML5AppNode(
+        source_id=source_id,
+        title=title,
+        license=licenses.CC_BY,
```

Comment:
Consider replacing this line with `license=get_license(licenses.CC_BY, copyright_holder='African Storybook...')`